usecases/ingestion_usecase.go

```diff
@@ -40,17 +44,63 @@ type IngestionUseCase struct {
 	GcsIngestionBucket  string
 }
 
-func (usecase *IngestionUseCase) IngestObjects(ctx context.Context, organizationId string, payloads []models.ClientObject, table models.Table) error {
+func (usecase *IngestionUseCase) IngestObjects(
+	ctx context.Context,
+	organizationId string,
+	objectType string,
+	objectBody json.RawMessage,
+) (int, error) {
+	logger := utils.LoggerFromContext(ctx)
+	tracer := utils.OpenTelemetryTracerFromContext(ctx)
+	ctx, span := tracer.Start(
+		ctx,
+		"IngestionUseCase.IngestObjects",
+		trace.WithAttributes(attribute.String("objectType", objectType)),
+		trace.WithAttributes(attribute.String("organizationId", organizationId)))
+	defer span.End()
+
 	if err := usecase.enforceSecurity.CanIngest(organizationId); err != nil {
-		return err
+		return 0, err
+	}
+
+	exec := usecase.executorFactory.NewExecutor()
+	dataModel, err := usecase.dataModelRepository.GetDataModel(ctx, exec, organizationId, false)
+	if err != nil {
+		return 0, errors.Wrap(err, "error getting data model in IngestObjects")
 	}
 
+	tables := dataModel.Tables
+	table, ok := tables[objectType]
+	if !ok {
+		return 0, errors.Wrapf(
+			models.NotFoundError,
+			"table %s not found in data model in IngestObjects", objectType,
+		)
+	}
+
+	parser := payload_parser.NewParser()
+	payload, validationErrors, err := parser.ParsePayload(table, objectBody)
+	if err != nil {
+		return 0, errors.Wrapf(
+			models.BadParameterError,
+			"Error while validating payload in IngestObjects: %v", err,
+		)
+	}
+	if len(validationErrors) > 0 {
+		encoded, _ := json.Marshal(validationErrors)
+		logger.InfoContext(ctx, fmt.Sprintf("Validation errors on IngestObjects: %s", string(encoded)))
+		return 0, errors.Wrap(models.BadParameterError, string(encoded))
+	}
+
+	var nb int
 	ingestClosure := func() error {
 		return usecase.transactionFactory.TransactionInOrgSchema(ctx, organizationId, func(tx repositories.Executor) error {
-			return usecase.ingestionRepository.IngestObjects(ctx, tx, payloads, table)
+			var err error
+			nb, err = usecase.ingestionRepository.IngestObjects(ctx, tx, []models.ClientObject{payload}, table)
+			return err
```

Comment:
For my curiosity, this syntax ensure we redeclare a `var err error` specitific variable inside the scope of the inline `func` that do not rely on the "outter scope" err ?

Reply:
Hmm, well yes, though in fact the line 98 is useless here (I removed it), because there already exists a variable `err` higher up in the function.
What I don't want specifically is to use `:=` line 99 because I want to update the variable `nb`
defined outside of the inline function. So if there was no variable err higher in the IngestObjects method, I may be tempted to do `nb, err:= ...` which would be wrong because it would define a new `nb` that is not the outer scope `nb`